explorer/explorer.go

```diff
@@ -135,9 +140,50 @@ func (exp *explorerUI) rootWebsocket(w http.ResponseWriter, r *http.Request) {
 				exp.wsHub.HubRelay <- sigPingAndUserCount
 			}
 		}()
-
+		go func() {
+			for {
+				msg := &WebSocketMessage{}
+				websocket.JSON.Receive(ws, &msg)
+				var webData WebSocketMessage
+				switch msg.EventId {
+				case "decodetx":
+					webData.EventId = "decodedtx"
+					log.Debug("Recieved decodetx signal for hex: ", msg.Messsage)
+					tx := exp.blockData.DecodeRawTransaction(msg.Messsage)
+					if tx != nil {
+						buff := new(bytes.Buffer)
+						enc := json.NewEncoder(buff)
+						enc.SetIndent("", "    ")
+						enc.Encode(tx)
+						webData.Messsage = buff.String()
+					} else {
+						log.Debugf("Could not decode raw tx")
+						webData.Messsage = fmt.Sprintf("Error: Could not decode hex %s", msg.Messsage)
+					}
+				case "sendtx":
+					webData.EventId = "senttx"
+					log.Debug("Recieved sendtx signal for hex: ", msg.Messsage)
+					txid, err := exp.blockData.SendRawTransaction(msg.Messsage)
+					if err != nil {
+						webData.Messsage = fmt.Sprintf("Error: %v", err)
+					} else {
+						webData.Messsage = fmt.Sprintf("Transaction sent: %s", txid)
+					}
+				}
+				if webData.EventId != "" {
```

Comment:
It's a matter of style, but I prefer to minimize indent.  That is,

```go
if webData.EventID == "" {
    continue
}
```

However, this seems like something that should be in the `default:` section of the `switch`